packages/blocks/src/lib/slam-map-2d/axes-helper.svelte

```diff
@@ -0,0 +1,48 @@
+<script lang="ts">
+import * as THREE from 'three';
+import { T } from '@threlte/core';
+import { Line2 } from 'three/examples/jsm/lines/Line2';
+import { LineGeometry } from 'three/examples/jsm/lines/LineGeometry';
+import { LineMaterial } from 'three/examples/jsm/lines/LineMaterial';
+
+export let length = 1;
+export let width = 0.2;
+
+const line = new Line2();
+const material = new LineMaterial();
+const geometry = new LineGeometry();
+const axes = ['red', 'green', 'blue'];
+const color = new THREE.Color();
+const colors = new Float32Array(27);
+
+for (const [index, axis] of axes.entries()) {
+  color.set(axis);
+
+  for (let j = index * 9; j < index * 9 + 9; j += 3) {
+    colors[j + 0] = color.r;
+    colors[j + 1] = color.g;
+    colors[j + 2] = color.b;
+  }
+}
```

Comment:
It doesn't look like `color` is used outside of this loop, and I had a little trouble tracking what was going on. Would this be easier to read?

```suggestion
const colors = new Float32Array(27);

for (const [index, axis] of axes.entries()) {
  const color = new THREE.Color(axis)

  for (let j = index * 9; j < index * 9 + 9; j += 3) {
    colors[j + 0] = color.r;
    colors[j + 1] = color.g;
    colors[j + 2] = color.b;
  }
}
```

It's also a little hard to understand what this `Float32Array` represents, why there's 27 elements, and where the indices `3`, `13`, and `23` below are coming from. I think moving the logic portion of this component to a unit-tested file would provide a lot of clarity

Reply:
Fair enough! Honestly, I don't hate an unrolled loop in cases like this, if you think it would help readability